### DISCUSSION TITLE
How to upgrade quarto cli using conda ?

### DISCUSSION THREAD

**Discussion by sunvitD** [quarto-dev/quarto-cli#11716] [Q&A]
### Description

For some reason `Quarto cli` was never installed on my windows machine through `pip` may be because of Anaconda. 

Error:
```
      copying build\lib\quarto_cli\share\deno_std\cache\deps\https\cdn.skypack.dev\010ab5a2e2b8bd9419eceeb3654e9d2f1412afd9065e89a8d3d03c58f29970dd.metadata.json -> build\bdist.win-amd64\wheel\.\quarto_cli\share\deno_std\cache\deps\https\cdn.skypack.dev
      error: could not create 'build\bdist.win-amd64\wheel\.\quarto_cli\share\deno_std\cache\deps\https\cdn.skypack.dev\010ab5a2e2b8bd9419eceeb3654e9d2f1412afd9065e89a8d3d03c58f29970dd.metadata.json': No such file or directory   
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for quarto-cli
Failed to build quarto-cli
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (quarto-cli)
```

So i finally installed quarto using `conda` with command `conda install conda-forge::quarto` but the installed version I have is `1.5.57` and even on upgrade it stays the same.

Can it be upgraded using `conda` command or do I need to find a solution by uninstalling anaconda and quarto and then reinstalling quarto cli ?

**Comment by mcanouil** [comment]
The Conda version is not officially packaged by the Quarto Team.
I believe the recipe should automatically update to the latest stable or might require a manual intervention on their repository.

https://github.com/conda-forge/quarto-feedstock

**Reply by sunvitD** [reply]
Thanks for the reply.

**Comment by cderv** [comment]
> For some reason Quarto cli was never installed on my windows machine through pip may be because of Anaconda.

We have some report in previous discussion about `pip install quarto-cli` not working with conda. 

For conda the conda-forge is indeed the one to use 

> The Conda version is not officially packaged by the Quarto Team. I believe the recipe should automatically update to the latest stable or might require a manual intervention on their repository.

We help on this bundle, but the release still needs to be done :
- https://github.com/conda-forge/quarto-feedstock/pull/65

It happens there is bundling problem that needs to be solved. So there is a delay on the release unfortunately.

**Reply by sunvitD** [reply]
Thank you for the reply. I am eagerly waiting for its release as the 1.5 series versions doesn't support branding/logo and many other features. It will be great to have the 1.6+ versions.

**Comment by sunvitD** [comment]
I have tried again to **install** quarto-cli using uv pip this time `uv pip install 'quarto-cli==1.6.37'` and there is some new information in the `logs` that I have not seen before when I used just `pip` to install.

Sharing the logs Below:
[quarto_install_issue_uv_log.txt](https://github.com/user-attachments/files/18209213/quarto_install_issue_uv_log.txt)

If there is an easy fix to this then please let me know.

Thanks !!!

**Reply by cderv** [reply]
Thanks a lot. This is quite interesting - thanks for sharing !

To be clear, you tried to do `uv pip install quarto-cli==1.6.37` inside a conda environment and **uv** produced this log ?

**Reply by sunvitD** [reply]
Glad to share, Yes this is exactly the case.

**Reply by sunvitD** [reply]
I am not sure if this can be relevant but I have also noticed that even when within a `virtual environment` Quarto sometimes uses another `Python` outside the virtual env. This was happening when I was trying to run Quarto using a `subprocess` within a `streamlit app` in `streamlit cloud`.

This is the [discussion link](https://discuss.streamlit.io/t/filenotfounderror-what-is-the-location-of-file-that-is-written-using-streamlit-script-hosted-on-streamlit-cloud-subprocess-using-written-file-gives-error-on-streamlit-cloud-but-works-on-local-machine/87610/12) with the streamlit dev team. 

And as per them `"quarto is running the wrong python"` `"problem is that quarto runs python code using the system python environment instead of the virtual environment."` 

And eventually he made it work by `"by creating a bash script that activates the venv and then runs quarto."`.

Not sure if this is helpful as I am not a software dev and do not have that technical skills but thought this could also be related so sharing it here.

**Reply by cderv** [reply]
We discuss using virtual environment at https://quarto.org/docs/projects/virtual-environments.html

About venv we say 
> To use the environment you need to activate it.

`quarto` will know that the environment is activated check environment variable and use this python version in priority. When the virtual environmnent is not activated, `quarto` python detection will follow the usual logic and find first python version having the desired kernel. So it could check in PATH and use the system one. 

> I have also noticed that even when within a virtual environment Quarto sometimes uses another Python outside the virtual env.

So did this happened in an activated environment ? Or otherwise what are you calling "within a virtual env" ? 

Do you mean installing `quarto-cli` inside a virtual env and then you would expect that specific version to use the same virtualenv it is installed in without having to activae the venv first ? 

I am trying to clarify how you think it should work. You can also look at python detection algorithm in our code base. 

Thanks !
